### PR TITLE
test(testing): add a test to ensure fakeAsync works with angular2/tes…

### PR DESCRIPTION
**Pull request by juliemr**
…ting

### CODE REVIEW

**Comment by asvifi85** [comment]
How to write jasmine test case for the following class having constructor which is having dependency on http?
jasmineerror

import {Injectable} from 'angular2/core';
import {Http, HTTP_PROVIDERS} from 'angular2/http';

@Injectable()
export class MockUserService {
    items:Array<any>;

    constructor(http:Http){
        http.get('http://127.0.0.1:8080/src/data/names.json')
        .subscribe(res => {
            this.items = res;
            console.log('results found');
        })
    }
}

it('Testing user login', inject([MockUserService,Http], (mockUserService:MockUserService ) => {
   let http:Http;
        let mockUserService: MockUserService = new MockUserService(http);
     expect(1+1).toEqual(2);

  });
  );

**Comment by antonybudianto** [comment]
@asvifi85 
in test you don't create (`new`-ing) the service, just inject it using `inject` method
here is for example:
```javascript
import {
    it,
    inject,
    describe,
    expect,
    beforeEach,
    beforeEachProviders
} from 'angular2/testing';
import {
    HTTP_PROVIDERS
} from 'angular2/http';
import { LoggerService } from './logger.service';

describe('AppComponent', () => {
    beforeEachProviders(() => [HTTP_PROVIDERS, LoggerService]);
    beforeEach(() => {
        spyOn(console, 'log');
    });

    it('should log successfully', inject([LoggerService], (logger) => {
        let message = 'Hello';
        logger.log(message);
        expect(console.log).toHaveBeenCalledWith(message);
    }));
});

```

`logger.service.ts` file:
```javascript
import {Injectable} from 'angular2/core';
import { Http } from 'angular2/http';

@Injectable()
export class LoggerService {
    constructor(private http: Http) {
    }

    log(message: string): void {
        console.log(message);
    }
}

```

**Comment by asvifi85** [comment]
How to inject Http dependecy into class in 'it' block.?

**Comment by antonybudianto7** [comment]
```javascript
it('should log successfully', inject([LoggerService, Http], ...
```
as long as the provider already registered in `beforeEachProviders`

**Comment by juliemr** [comment]
@vsavkin for fakeAsync stuff.

File: modules/angular2/test/testing/testing_public_spec.ts [hunk 1]
```diff
@@ -168,6 +169,13 @@ export function main() {
                (value) => { expect(value).toEqual('async value'); });
          }));
 
+      it('should allow the use of fakeAsync',
+         inject([FancyService], fakeAsync((service) => {
+                  var value;
+                  service.getAsyncValue().then(function(val) { value = val; }) tick();
```
Comment (kegluneq) [inline modules/angular2/test/testing/testing_public_spec.ts]:
```bash
[17:42:37] Error: [DiffingTSCompiler]: Typescript found the following errors:
  angular2/test/testing/testing_public_spec.ts (175, 80): ';' expected.
```

**Comment by kegluneq** [comment]
LGTM when LGT Travis (modulo flaky / broken js.dev task)

File: modules/angular2/test/testing/testing_public_spec.ts (hunk 1 above)
Comment (juliemr) [inline modules/angular2/test/testing/testing_public_spec.ts]: thanks! done.

**Comment by vsavkin** [comment]
Merged via https://github.com/angular/angular/commit/aaafdf03ce744860d6a1efcad7c0376d1a25221f

**Pull request closed without merging**